pytket/phir/phirgen_parallel.py

```diff
@@ -0,0 +1,350 @@
+                    qubits2groups[qubit] = group_number
+                else:
+                    # make a new group:
+                    if gate == tk.OpType.Rz:
+                        rz_group_number += 3
+                        group_number = rz_group_number
+                    elif gate == tk.OpType.PhasedX:
+                        r1xy_group_number += 3
+                        group_number = r1xy_group_number
+                    else:
+                        other_group_number += 3
+                        group_number = other_group_number
+                    groups[group_number] = [sc]
+                    qubits2groups[qubit] = group_number
+        index += 1
+
+    groups = dict(sorted(groups.items()))
+    return groups  # noqa: RET504
+
+
+def groups2qops(groups: dict[int, list[tk.Command]], ops: list[dict[str, Any]]) -> None:  # noqa: PLR0912
+    """Convert the groups of parallel ops to properly formatted PHIR."""
+    for group_number in groups:
+        group = groups[group_number]
+        angles2qops: dict[tuple, dict[str, Any]] = {}  # type: ignore[type-arg]
+        comment_insert = None
+        for qop in group:
+            if not qop.op.is_gate():
+                append_cmd(qop, ops)
+            else:
+                comment_insert = str(qop.op)
+                arg: UnitID | list[Qubit] = (
+                    qop.qubits[0] if len(qop.qubits) == 1 else qop.qubits
+                )
+                angles = qop.op.params
+                if tuple(angles) not in angles2qops:
+                    fmt_qop: dict[str, Any] = {
+                        "qop": tket_gate_to_phir[qop.op.type],
+                        "angles": [angles, "pi"],
+                    }
+                    if len(qop.qubits) == 1:
+                        fmt_qop["args"] = [arg_to_bit(arg)]  # type: ignore[arg-type]
+                    else:
+                        arg = [arg_to_bit(a) for a in arg]  # type: ignore[misc, union-attr]
+                        fmt_qop["args"] = [arg]
+                    angles2qops[tuple(angles)] = fmt_qop
+                else:
+                    fmt_qop = angles2qops[tuple(angles)]
+                    if len(qop.qubits) == 1:
+                        fmt_qop["args"].append(arg_to_bit(arg))  # type: ignore[arg-type]
+                    else:
+                        arg = [arg_to_bit(a) for a in arg]  # type: ignore[misc, union-attr]
+                        fmt_qop["args"].append(arg)
+        # in the case where the qop is not a gate (a conditional for example)
+        # this branch is skipped because non-gate sub-commands
+        # are always the only member of their group (see process_sub_commands)
+        if len(angles2qops) > 1:
+            pll_block: dict[str, Any] = {"block": "qparallel", "ops": []}
+            for phir_qop in angles2qops.values():
+                pll_block["ops"].append(phir_qop)
+            comment = {"//": f"Parallel {comment_insert}"}
```

Comment:
What's the goal of this comment? I see only the last op being commented currently, e.g.:
```py
[...]
),
Comment(c='Parallel PhasedX(0.5, 0.5)'),
QParBlock(
    metadata=None,
    block='qparallel',
    ops=[
        SQOp(metadata=None, qop='R1XY', angles=([0.09971088036499957, 0.5], 'pi'), args=[('q', 7)]),
        SQOp(metadata=None, qop='R1XY', angles=([0.01941756381009971, 0.5], 'pi'), args=[('q', 9)]),
        SQOp(metadata=None, qop='R1XY', angles=([0.18839670705078193, 0.5], 'pi'), args=[('q', 19)]),
        SQOp(metadata=None, qop='R1XY', angles=([0.05780952798587828, 1.5], 'pi'), args=[('q', 1)]),
        SQOp(metadata=None, qop='R1XY', angles=([0.266222066514598, 1.5], 'pi'), args=[('q', 14)]),
        SQOp(metadata=None, qop='R1XY', angles=([0.1035418971101575, 0.5], 'pi'), args=[('q', 4)]),
        SQOp(metadata=None, qop='R1XY', angles=([0.17261960632297724, 0.5], 'pi'), args=[('q', 12)]),
        SQOp(metadata=None, qop='R1XY', angles=([0.18491715565571812, 0.5], 'pi'), args=[('q', 0)]),
        SQOp(metadata=None, qop='R1XY', angles=([0.036800581240582816, 0.5], 'pi'), args=[('q', 5)]),
        SQOp(metadata=None, qop='R1XY', angles=([0.07822775603170962, 0.5], 'pi'), args=[('q', 11)]),
        SQOp(metadata=None, qop='R1XY', angles=([0.5, 0.5], 'pi'), args=[('q', 15), ('q', 13), ('q', 10), ('q', 17), ('q', 3), ('q', 16), ('q', 18), ('q', 6), ('q', 2), ('q', 8)])
    ]
),

```